app/services/force/listing_service.rb

```diff
@@ -74,10 +74,14 @@ def self.translations_invalid?(listing_translations)
     end
 
     def self.translations_are_outdated?(cache_last_modified, listing_last_modified)
+      return true if cache_last_modified.nil? || listing_last_modified.nil?
+
       parse_time(cache_last_modified) < parse_time(listing_last_modified)
     end
 
     def self.listing_is_outdated?(cache_last_modified, listing_last_modified)
+      return true if cache_last_modified.nil? || listing_last_modified.nil?
```

Comment:
can these checks happen outside of the functions? What do we actually want to happen if the cache_last_modified is nil or the listing_last_modified is nil?

Reply:
We would want to rerun translations so that the LastModifiedDate is populated. If the listing LastModifiedDate is nil, then it would make since to refresh the listing. That would be a rare case, and it would only happen if the listing is undefined for some reason, which should already be handled.

I think we should keep these checks within these functions

Reply:
If they stay in the function, then it is confusing. What if the listing_is_outdated function is run and cache_last_modified is nil but listing_last_modified is not nil. Then we are automaticlaly returning that listing_is_outdated is true, but that isn't the case

Reply:
I think if any parameter of the comparison is invalid, we should default to returning `true` for invalid. 
I added more explicit error handling to illustrate that. 
This will report the error but not prevent the page from loading. And should ensure that the translations will be reprocessed if needed.